data-node/src/main/java/org/graylog/datanode/bindings/PreflightChecksBindings.java

```diff
@@ -47,13 +60,20 @@ protected void configure() {
         addPreflightCheck(DatanodeDirectoriesLockfileCheck.class);
         addPreflightCheck(OpenSearchPreconditionsCheck.class);
         addPreflightCheck(OpensearchDataDirCompatibilityCheck.class);
+        addPreflightCheck(PasswordSecretPreflightCheck.class);
+
+        bindLimittedObjectMapper();
 
         // Mongodb is needed for legacy datanode storage, where we want to extract the certificate chain from
         // mongodb and store it in local keystore
         bind(MongoConnection.class).toProvider(MongoConnectionProvider.class);
         addPreflightCheck(DatanodeKeystoreCheck.class);
     }
 
+    private void bindLimittedObjectMapper() {
```

Comment:
I would move these to the `PreflightClusterConfigurationModule`

Reply:
makes sense, done